docs/source/examples/loschmidt_echo_revival_zne.md

```diff
@@ -0,0 +1,396 @@
+    noise_model = NoiseModel(basis_gates)
+    # Add depolarizing noise to the 1-qubit gates in the basis set
+    noise_model.add_all_qubit_quantum_error(
+        depolarizing_error(noise_level, 1), basis_gates[:3]
+    )
+    
+    echo = np.array([
+        simulate_noisy(
+            create_circuit(params.dt, k, n_qubits=params.n_qubits, measure=True),
+            noise_model,
+            params.n_shots,
+            scale_factor
+        )
+        for k in range(0, params.M + 1)
+    ])    
+
+    return params.t_vals, echo
+```
+
+```{code-cell} ipython3
+result_depolarizing = run_depolarizing_noise(noise_level=0.0005)
+```
+
+```{code-cell} ipython3
+setup_plot("Ideal and noisy simulations")
+legend = []
+add_to_plot(*result_ideal, "ideal", legend)
+add_to_plot(*result_depolarizing, "depolarizing noise, $p = 0.0005$", legend)
+plt.legend(legend)
+plt.show()
+```
+
+As expected, the Loschmidt echo revival is weaker than in the ideal cases, and we get a lower peak. Applying gate folding suppresses the peak further.
+
+```{code-cell} ipython3
+scale_factors = [1, 2, 3]
+result_depolarizing_scaled = [
+    run_depolarizing_noise(noise_level=0.0005, scale_factor=alpha) for alpha in scale_factors
+]
+```
+
+```{code-cell} ipython3
+setup_plot("Scaled noisy simulations")
+legend = []
+add_to_plot(*result_ideal, "ideal", legend)
+for alpha, result in zip(scale_factors, result_depolarizing_scaled):
+    add_to_plot(*result, "$\\alpha = {}$".format(alpha), legend)
+plt.legend(legend)
+plt.show()
+```
+
+## Error mitigation with zero-noise extrapolation
+At this level of noise, we can use ZNE to mostly recover the ideal result. Running the noisy simulation is expensive (especially as the scale factor $\alpha$ increases), so rather than using the high-level functions for ZNE, we apply the static method `RichardsonFactory.extrapolate` to the results from the previous cell.
+
+```{code-cell} ipython3
+from mitiq.zne.inference import RichardsonFactory
+
+result_zne = RichardsonFactory.extrapolate(scale_factors,
+                                           [r[1] for r in result_depolarizing_scaled])
+```
+
+```{code-cell} ipython3
+setup_plot("ZNE for depolarizing noise model, $p = 0.0005$")
+legend = []
+add_to_plot(*result_ideal, "ideal", legend)
+for alpha, result in zip(scale_factors, result_depolarizing_scaled):
+    add_to_plot(*result, "$\\alpha = {}$".format(alpha), legend)
+add_to_plot(result_depolarizing_scaled[0][0], result_zne, "mitigated", legend)
+plt.legend(legend)
+plt.show()
+```
+
+Increasing the baseline noise level makes it much harder to reconstruct the peak with ZNE.
+
+```{code-cell} ipython3
+scale_factors = [1, 2, 3]
+result_depolarizing_scaled = [
+    run_depolarizing_noise(noise_level=0.005, scale_factor=alpha) for alpha in scale_factors
+]
+```
+
+```{code-cell} ipython3
+result_zne = RichardsonFactory.extrapolate(scale_factors,
+                                           [r[1] for r in result_depolarizing_scaled])
```

Comment:
nit: we don't have strict coding style guidelines in Mitiq, but especially in tutorials, I am not a fan of this identation, because part of the code ends up being hidden at the right of the cell. Imo the following is more readable:

```suggestion
result_zne = RichardsonFactory.extrapolate(
    scale_factors,
    [r[1] for r in result_depolarizing_scaled]
)
```

Reply:
Done in 454a6be